### PR TITLE
[Delete Response Confirmation] Added delete confirmation popup for email response and task

### PR DESCRIPTION
**Pull request by fnormand01**
# Description

I used the _PopupBox.jsx_ and _SystemMessage.jsx_ components to create a delete confirmation dialog when you want to delete an email response and / or a task.

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshot

**Email Response Delete Confirmation:**

image

image

**Task Delete Confirmation:**

image

image

# Testing

Manual steps to reproduce this functionality:

1.  Start the eMIB Sample Test.
2.  Complete the _pre-test_ process.
3.  Open the _Inbox_ tab.
4.  Add an email response and a task.
5.  Try to delete both of them and make sure that the popup dialog is looking good and working well, meaning that it does what it is supposed to do.

# Checklist

Applicable for all code changes.

- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [ ] ~~I have made corresponding changes to the documentation~~
- [x] My changes generate no new compiler warnings
- [ ] ~~My changes generate no new accessibility errors and/or warnings~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [ ] ~~I have researched WCAG2.1 accessibility standards and met them in this PR (can be navigated using a keyboard)~~
- [x] My changes look good on IE 10+, Firefox, and Chrome

### CODE REVIEW

File: frontend/src/text_resources.js [hunk 1]
```diff
@@ -369,6 +369,14 @@ let LOCALIZE = new LocalizedStrings({
         taskContent: {
           task: "Your task(s):",
           reasonsForAction: "Your reasons for action:"
+        },
+        deleteResponseConfirmation: {
+          title: "Are you sure you want to delete this response?",
+          systemMessageTitle: "Warning!",
```
Comment (MichaelCherry) [inline frontend/src/text_resources.js]: A bit nitpicky, but I think "Warning!" would be something we would use more than once. Should this be under commons?
We could also just move it if and when we create another dialog that also needs it. I'm good with either.

File: frontend/src/text_resources.js [hunk 2]
```diff
@@ -841,6 +849,14 @@ let LOCALIZE = new LocalizedStrings({
         taskContent: {
           task: "FR Your task(s):",
           reasonsForAction: "FR Your reasons for action:"
+        },
+        deleteResponseConfirmation: {
+          title: "FR Are you sure you want to delete this response?",
+          systemMessageTitle: "Avertissement!",
```
Comment (MichaelCherry) [inline frontend/src/text_resources.js]: see previous

File: frontend/src/text_resources.js (hunk 1 above)
Comment (caleybrock) [reply frontend/src/text_resources.js]: I think it's fine to have duplicated strings - it prevents us from changing it and causing problems in other places.

File: frontend/src/components/eMIB/ActionViewEmail.jsx [hunk 3]
```diff
@@ -142,10 +153,37 @@ class ActionViewEmail extends Component {
...
+                  />
+                </div>
+                <div>{LOCALIZE.emibTest.inboxPage.deleteResponseConfirmation.description}</div>
+              </div>
+            }
+            leftButtonType={BUTTON_TYPE.danger}
+            leftButtonTitle={LOCALIZE.emibTest.inboxPage.emailCommons.deleteButton}
+            leftButtonAction={() => this.props.deleteEmail(this.props.emailId, this.props.actionId)}
+            rightButtonType={BUTTON_TYPE.primary}
+            rightButtonTitle={LOCALIZE.commons.returnToTest}
```
[the remark this answers is not in the document]
Comment (fnormand01) [reply frontend/src/components/eMIB/ActionViewEmail.jsx]: By default, if there is no action, the button will trigger the close action, so it's not needed.

File: frontend/src/components/eMIB/ActionViewTask.jsx [hunk 4]
```diff
@@ -67,10 +78,37 @@ class ActionViewTask extends Component {
...
+                  />
+                </div>
+                <div>{LOCALIZE.emibTest.inboxPage.deleteResponseConfirmation.description}</div>
+              </div>
+            }
+            leftButtonType={BUTTON_TYPE.danger}
+            leftButtonTitle={LOCALIZE.emibTest.inboxPage.emailCommons.deleteButton}
+            leftButtonAction={() => this.props.deleteTask(this.props.emailId, this.props.actionId)}
+            rightButtonType={BUTTON_TYPE.primary}
+            rightButtonTitle={LOCALIZE.commons.returnToTest}
```
[the remark this answers is not in the document]
Comment (fnormand01) [reply frontend/src/components/eMIB/ActionViewTask.jsx]: By default, if there is no action, the button will trigger the close action, so it's not needed.

File: frontend/src/text_resources.js (hunk 1 above)
Comment (MichaelCherry) [reply frontend/src/text_resources.js]: Ok. I'll resolve this then

Comment (fnormand01) [reply frontend/src/text_resources.js]: The other system messages have their own titles, like the following:

image

So I'm not sure that we'll use the word 'warning' as often as we think.

Comment (fnormand01) [reply frontend/src/text_resources.js]: We'll keep it that way for now and if we use this word more often, then we can add it to the _commons_ words.

File: frontend/src/text_resources.js (hunk 2 above)
Comment (fnormand01) [reply frontend/src/text_resources.js]: Same as my previous comment...

**Comment by fnormand01** [comment]
To fix the tests issues, I simply deleted the **'delete button is triggered'** tests, since these functions are now called inside the _PopupBox.jsx_ component.

Also, there are tests in this _PopupBox.jsc_ component that makes sure that the button actions are called properly.

**Pull request merged**